contracts/test/integration/IntegrationTestMaplePCVDeposit.t.sol

```diff
@@ -0,0 +1,643 @@
+        usdc.transfer(address(this), targetUsdcBalance);
+
+        // Warp forward and make payment
+        vm.warp(block.timestamp + gracePeriod + paymentInterval);
+        usdc.approve(mapleLoan, type(uint256).max);
+        (uint256 principal1, uint256 interest1) = IMapleLoan(mapleLoan)
+            .makePayment(7_000e6);
+        assertEq(principal1, 0);
+        assertTrue(interest1 > 5_900e6);
+
+        // Pool Manager claims the interests
+        vm.prank(MainnetAddresses.MPL_ORTHOGONAL_POOL_DELEGATE);
+        IMaplePool(maplePool).claim(mapleLoan, mapleDebtLockerFactory);
+
+        // Accrue interests in PCVDeposit and check live+recorded balances
+        uint256 protocolEarnedInterests = usdcDeposit.balance() -
+            usdcDeposit.lastRecordedBalance();
+        assertTrue(protocolEarnedInterests > 1e6); // ast least 1 USDC of interests
+
+        // check Harvest event for USDC interests
+        vm.expectEmit(true, false, false, true, address(usdcDeposit));
+        emit Harvest(
+            MainnetAddresses.USDC,
+            int256(protocolEarnedInterests),
+            block.timestamp
+        );
+        usdcDeposit.accrue();
+
+        assertEq(
+            usdcDeposit.lastRecordedBalance(),
+            targetUsdcBalance + protocolEarnedInterests
+        );
+        assertEq(usdcDeposit.lastRecordedBalance(), usdcDeposit.balance());
+    }
+
+    function testAccrueLossesIncurredButCovered() public {
+        assertEq(usdcDeposit.lastRecordedBalance(), targetUsdcBalance);
+
+        // Someone creates a loan
+        address mapleLoanFactory = 0x36a7350309B2Eb30F3B908aB0154851B5ED81db0;
+        address mapleDebtLockerFactory = 0xA83404CAA79989FfF1d84bA883a1b8187397866C;
+        address mapleLoan;
+        uint256 gracePeriod = 432000; // 5 days
+        uint256 paymentInterval = 2592000; // 30 days
+        bytes memory arguments = abi.encode(
+            address(this), // borrower
+            address(0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599), // assets[0] = collateral = WBTC
+            MainnetAddresses.USDC, // assets[1] = principal = USDC
+            gracePeriod, // terms[0]
+            paymentInterval, // terms[1] = paymentInterval
+            2, // terms[2] = numberOfPayments
+            0, // amounts[0] = collateralRequired
+            targetUsdcBalance, // amounts[1] = principalRequested
+            targetUsdcBalance, // amounts[2] = endingPrincipal
+            0.12e18, // rates[0] = interestRate = 12%
+            0.02e18, // rates[1] = earlyFeeRate = 2%
+            0, // rates[2] = lateFeeRate = 0
+            0.05e18 // rates[3] = lateInterestPremium = 5%
+        );
+        mapleLoan = IMapleLoanFactory(mapleLoanFactory).createInstance(
+            arguments,
+            keccak256(
+                bytes("some salt for testAccrueLossesIncurredButCovered()")
+            )
+        );
+        vm.label(mapleLoan, "mapleLoan");
```

Comment:
feels like tests could be made cleaner by breaking this logic out into a helper function to reduce size